### PR TITLE
Cosmosim latest unittests

### PR DESCRIPTION
**Pull request by agroener**
Picking up where I left off regarding the creation of tests for the CosmoSim module. This PR is ideally going to replace a previous one (https://github.com/astropy/astroquery/pull/458) once it is completely up and running, ideally. This is also related to a current issue (https://github.com/astropy/astroquery/issues/599).

### CODE REVIEW

File: astroquery/cosmosim/core.py [hunk 1]
```diff
@@ -379,8 +387,8 @@ def check_all_jobs(self, phase=None, regex=None, sortby=None):
             if not phase and not regex:
                 if not sortby:
                     t = Table()
-                    t['JobID'] = self.job_dict.keys()
-                    t['Phase'] = self.job_dict.values()
+                    t['JobID'] = [i for i in self.job_dict.keys()]
```
Comment (keflavich) [inline astroquery/cosmosim/core.py]: could use `list(self.job_dict.keys())` for the same effect

File: astroquery/cosmosim/tests/test_cosmosim.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+try:
+    from ...cosmosim import CosmoSim
+    COSMOSIM_IMPORTED = True
+except:
+    COSMOSIM_IMPORTED = False
+from ...exceptions import LoginError
+SKIP_TESTS = not(HAS_KEYRING and COSMOSIM_IMPORTED)
+
+@pytest.mark.skipif('SKIP_TESTS')
+@remote_data
```
Comment (keflavich) [inline astroquery/cosmosim/tests/test_cosmosim.py]: since this is remote, it should go into `the test_cosmosim_remote` file

**Comment by keflavich** [comment]
@agroener your testing issues are arising because you `import ipdb`. It seems that ipdb and pytest are incompatible

File: astroquery/cosmosim/core.py (hunk 1 above)
Comment (agroener) [inline astroquery/cosmosim/core.py]: Ah, yes. I had this in another branch but forgot to add it here. Oddly enough, casting to a list fails. I had to actually do a list comprehension to fix this.

**Comment by agroener** [comment]
Ok. I've written some tests for the login/logout procedure for cosmosim. I've pushed up my changes because at this point in time, keyring is now doing something unexpected (since the last time I used it). It now is asking me for:

    Please enter password for encrypted keyring:

To remind ourselves, cosmosim uses the <code>login</code> method from its base class <code>QueryWithLogin</code>. Either keyring has changed in some way, or the implementation of this method has. Any thoughts?

**Comment by keflavich** [comment]
`keyring` requires you to enter your global password to get access to any hidden passwords.  On Mac OS, this is done with the OS Keychain app, and on linux it's done with something called 'secret service': https://pypi.python.org/pypi/keyring.  So I think that message is most likely coming from secret service and you're on a linux box?  Otherwise I don't know what it means exactly

**Comment by agroener** [comment]
Okay, that makes a bit more sense now. I did most of my developing on OS X previously. Now, I'm on linux, which is why the expected keyring behavior is differing. I'll go back to OS X for now, but I need to normalize the behavior after I get all of this working. I don't want this module to be platform dependent in any way.

**Comment by keflavich** [comment]
@agroener where do we stand with this and #458?

**Comment by agroener** [comment]
Well, I had intended to complete both a few weeks ago, but once I began working, I realized there were some minor discrepancies with keyring between OS X and Linux which was causing some unexpected login behavior. I've since fixed the issue, and things seem to be normalized a bit. The tests which I spent time on this last round of coding were regarding login/logout, of which there are a number of possible combinations when storing/deleting credentials are considered. 

I'll see if I can get around to finishing off login/logout and query tests in the next couple of weeks, pending my work schedule.

**Comment by keflavich** [comment]
This has lingered for a little more than a week.  I'm closing it.  Please reopen if you complete it.

**Pull request closed without merging**